### PR TITLE
#69 Implement with UAST

### PR DESCRIPTION
**Pull request by ifrankwang**

### CODE REVIEW

**Review by filiphr — CHANGES_REQUESTED** [review]
Thanks a lot for the PR @ifrankwang. You saved me quite some time 😄 .

Can you please let me know why we need to change the language for the reference contributors in the plugin xmls?

File: build.gradle [hunk 1]
```diff
@@ -32,7 +32,7 @@ repositories {
 }
 
 intellij {
-    version System.getenv().getOrDefault('IDEA_VERSION', ideaVersion)
+    version System.getenv().getOrDefault('IDEA_VERSION', '2020.3.3')
```
[the remark this answers is not in the document]
Comment (ifrankwang) [reply build.gradle:35]: Sure, I'll change it back.

**Comment by ifrankwang** [comment]
> Thanks a lot for the PR @ifrankwang. You saved me quite some time 😄 .
> 
> Can you please let me know why we need to change the language for the reference contributors in the plugin xmls?

@filiphr Because UAST would noly work in this way. Here's description in the offical document:
> To register extensions applicable to UAST, specify language="UAST" in plugin.xml.

**Comment by filiphr** [comment]
>Because UAST would noly work in this way. Here's description in the offical document:
>
>>To register extensions applicable to UAST, specify language="UAST" in plugin.xml.

I read that as well, but that would mean that those extensions are applied for all UAST languages. Which means that the `MapstructReferenceContributor` will be applied both for Java and Kotlin. However, I don't think that this is correct, since the `MapstructReferenceContributor` is Java specific (the element patterns are Java specific).

I tried your PR yesterday without using language `UAST` and everything works correctly. I'll integrate your PR without that change and will do a release

**Comment by ifrankwang** [comment]
Sure.

**Comment by filiphr** [comment]
I've manually merged this in 6942570183354afabbb109ec874ead116606d27f and also made it work for earlier (pre 2020.3) IntelliJ versions with a1ec88a8771206b337bfa35430cc0735d132e8ab.

Again thanks a lot for the help @ifrankwang

**Pull request closed without merging**